src/types.ts

```diff
@@ -1,5 +1,7 @@
 export type LogLevel = 'fatal' | 'error' | 'warn' | 'info' | 'config' | 'debug' | 'trace'
 
+type SpawnOpt = { [key: string]: string | string[] | SpawnOpt }
```

Comment:
Why do we define a custom type for this, shouldn't it be the Node.js type directly?

Reply:
the `spawn` argument is overloaded so the idea was kind of a lazy way to support the loose typing. Since we are using the `command`, `args`, then `options`, we could do an or on those types, which would look kind of wild since its every combination of `stdio` options for the 3 generics, which means 9 types plus the `SpawnOptionsWithoutStdio` type

```ts
type SpawnOpt = SpawnOptionsWithStdioTuple<StdioNull, StdioNull, StdioNull> | SpawnOptionsWithStdioTuple<StdioNull, StdioNull, StdioPipe> |  SpawnOptionsWithStdioTuple<StdioNull, StdioPipe, StdioNull> | SpawnOptionsWithStdioTuple<StdioPipe, StdioNull, StdioNull> | SpawnOptionsWithStdioTuple<StdioNull, StdioPipe, StdioPipe> | SpawnOptionsWithStdioTuple<StdioPipe, StdioNull, StdioPipe> | SpawnOptionsWithStdioTuple<StdioPipe, StdioNull, StdioPipe> |  SpawnOptionsWithStdioTuple<StdioPipe, StdioPipe, StdioNull> |  SpawnOptionsWithStdioTuple<StdioPipe, StdioPipe, StdioNull> | SpawnOptionsWithStdioTuple<StdioPipe, StdioPipe, StdioPipe> | SpawnOptionsWithoutStdio
```

We can probably simplify the above to something like this?

```
import type { SpawnOptionsWithoutStdio, SpawnOptionsWithStdioTuple, StdioPipe, StdioNull } from 'node:child_process'

type StdioOption = StdioNull | StdioPipe
type SpawnOpt = SpawnOptionsWithoutStdio | SpawnOptionsWithStdioTuple<StdioOption, StdioOption, StdioOption>
```

I'll commit this

Reply:
node types used in [cf95c7b](https://github.com/webdriverio-community/node-geckodriver/pull/552/commits/cf95c7ba8afe871520d835baa7795a3dfdec2629)

Reply:
ill update the docs as well assuming this change is ok

Reply:
Thanks for the clarification!